src/base/bittorrent/torrentinfo.cpp

```diff
@@ -105,21 +105,34 @@ TorrentInfo TorrentInfo::loadFromFile(const QString &path, QString *error) noexc
         return TorrentInfo();
     }
 
+    const qint64 fileSize = file.size();
     const qint64 fileSizeLimit = 100 * 1024 * 1024;  // 100 MB
-    if (file.size() > fileSizeLimit) {
+    if (fileSize > fileSizeLimit) {
         if (error)
             *error = tr("File size exceeds max limit %1").arg(fileSizeLimit);
+        file.close();
         return TorrentInfo();
     }
 
-    const QByteArray data = file.read(fileSizeLimit);
```

Comment:
IMO you should just/only wrap this line within `try ... catch` block, other changes are unnecessary and have minuscule effects.

Reply:
Then all other code using `data` should be inside this block too. Or `data` should be declared first outside of it.

Reply:
The intent was to try to separate and regain some control over the malloc step. If you look at QByteArray QIODevice::read(qint64 maxSize), it looks like it will always allocate at least 100MB if called with fileSizeLimit.

http://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/io/qiodevice.cpp

Reply:
Ah, another surprise from Qt... then `const QByteArray data = file.read(std::min(file.size(), fileSizeLimit));` should be sufficient.

Reply:
Go ahead.